### PR TITLE
Building.get_binaryen_tool improvements, followup to #9697

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -2874,16 +2883,20 @@ def get_binaryen_bin():
     return os.path.join(BINARYEN_ROOT, 'bin')
 
   @staticmethod
-  def get_binaryen_command(tool, args=[], debug=False):
+  def get_binaryen_command(tool, args=[], infile=None, outfile=None, debug=False):
```
Comment (sbc100) [inline tools/shared.py]: Why not make them mandatory?  (and come before the extra args?)

Comment (kripken) [reply tools/shared.py]: The output file should definitely be optional, as sometimes we don't pass it.

The input is currently passed all the time, so I can change that one if you want?

Comment (sbc100) [reply tools/shared.py]: Sounds good how about:

```
def get_binaryen_command(tool, infile, outfile=None, args=[], debug=False):
```

Comment (kripken) [reply tools/shared.py]: sounds good @sbc100, I rewrote it that way.

**Pull request merged**